Skeleton UI Framework

amirhmk opened the pull request:
This should give us an idea what `Skeleton` would look like, and how it can change the existing UI. https://github.com/misterpeddy/hands-down/issues/2

This tiny "boilerplate" gives us some of the basic UI stylings/components we may need, including buttons, tables, Grid, Typography, and overall simple enough UI utility. 

I wasn't really sure about the UX yet, just wanted to see how much this would give us. I specially like the fact that the buttons match the colour of the key points - would be cool to have that anyways!

I have moved the assets into the new directory `public`.  This includes `js` directory, and CSS stylesheets. (Including the newly added Skeleton stylesheet)

Berkmann18 commented:
The UI looks great so far.
A couple of UI/UX things that could be improved:
1. The buttons don't have different visual states (i.e. the same colour despite the changing text), so it would be nice to have them change colour (e.g. to a red shade or what would be _the red to green_ for that cyan shade)
2. The boolean button after "Label" deforms the space when it changes to "FALSE" (i.e. buttons being pushed to the next row), and it would be better to have it not deform the rest of the UI when its state changes.
3. the container could do with less white space (as it seems a bit too constrained and not taking advantage of the space) 
4. the layout doesn't seem responsive enough to the point that either the button/label structure looks weird (could be laid out in a column or row fashion) and on smaller screens, the camera view is overflowing the screen.

In any case, that's a great start, well done!

Berkmann18 commented on index.html:
```diff
@@ -15,18 +19,44 @@
+<link href="css/normalize.css" rel="stylesheet"/>
+<link href="css/skeleton.css" rel="stylesheet">
+</head>
+
```
Wouldn't it be better to indent those so that they are clearly _inside_ `<head> </head>`?
Same goes for the content _inside_ `<body></body>`.
It's not a problem per se, I just think it would make the code more readable.
Speaking about code readability, shall we use a formatter like Prettier and perhaps, a linter as well like ESLint?

amirhmk replied:
Have addressed in https://github.com/misterpeddy/hands-down/pull/18

misterpeddy commented on index.html:
```diff
@@ -15,18 +19,44 @@
+        <div class="row">
+          <canvas id="output" width="640" height="480" style="border: 1px solid black;">
+        </div>
+  </div>
```
Should these 2 be indented?

misterpeddy commented on index.html:
```diff
@@ -15,18 +19,44 @@
+          <canvas id="output" width="640" height="480" style="border: 1px solid black;">
+        </div>
+  </div>
+</div>
```
nit: indent

misterpeddy commented on app.js:
```diff
@@ -10,6 +10,7 @@ const MODELS_DIR = 'models';
 const MODELS_MOUNT_PATH = '/models';
 const INDEX_PATH = path.join(__dirname + '/index.html');
 
+app.use(express.static('public'));
```
3 things:
1. nit: plz pull literals into constants defined top of file.
2. what is public directory supposed to represent? what goes inside it vs. outside?
3. do we not need to change JS_DIR since we've moved it inside public?

I'm not 100% sure what best practices are for setting up assets directory with express so lmk if my questions are off mark :)

Berkmann18 replied:
By convention (at least regarding `express`), `/public` contains all the publically available files users will see (including the styles/script/images/... needed for the HTML pages) when using the web app.
A good example of in/out when it comes to express is:
![image](https://user-images.githubusercontent.com/8260834/77835845-ca124380-7148-11ea-911c-c36c016fc30a.png)
Source: https://expressjs.com/en/starter/generator.html

Since @amirhmk copied the `js/*.js` files to `public/js` and that those are used directly by `index.html`, the line with `app.use(JS_MOUNT_PATH, express.static(JS_DIR));` and both `JS_*` constants can be removed.

misterpeddy replied:
Excellent thank you for the explanation!

misterpeddy commented on public/css/normalize.css:
```diff
@@ -0,0 +1,444 @@
+/*! normalize.css v3.0.2 | MIT License | git.io/normalize */
```
Are normalize and skeleton copied from somewhere / generated from something? If we want to add additional styling, will we do it in one of these or do it in separate files? In general how easy is it to build upon?

amirhmk replied:
Yes they are downloaded from [Skeleton's](http://getskeleton.com) website. I would imagine we would not try to touch those stylesheets, and make our own if it's a septette styling. In general I think it depends on the scope, but I would say using `themes` and CSS in JS are the new thing now, but it's probably an overkill here. 

In general I imagine the app doesn't require too much customization and changes, so modifying the Skeleton CSS could be an option too, though not if we're going to heavily change it

misterpeddy replied:
Sounds good - since it's from an external source let's keep it as is and when we need to add new styling we'll make a separate stylesheet.

Berkmann18 commented on app.js:
```diff
@@ -4,13 +4,12 @@ const path = require('path');
 
-app.use(JS_MOUNT_PATH, express.static(JS_DIR));
+app.use(express.static(PUBLIC_DIR));
 app.use(MODELS_MOUNT_PATH, express.static(MODELS_DIR));
```
@misterpeddy @amirhmk I've applied Pedram's nitpick and removed what seemed redundant (the app doesn't need to access the js modules directly as `index.html` already includes `main.js` which imports the rest.
So unless there's an actual need to have the modules accessible on `http://.../js` then we can merge this and start the next iteration?

amirhmk replied:
ah amazing, thank you!

amirhmk commented:
Have merge `master` in and solved all listing issues as a result. Though seems like the `skeleton.css` files had a bunch of small errors, not worth fixing and not fixable automatically. So I have added a `style ignore` file so it can pass the pre-commit hook. 

Moved the export data section to its row, so at least there is no UI change (deformation), but we can definitely improve upon this in the next iteration!
